pkg/controller/whisker/whisker_controller.go

```diff
@@ -0,0 +1,192 @@
+		whisker.ManagedClusterConnectionSecretName,
+		render.ProjectCalicoAPIServerTLSSecretName(operatorv1.TigeraSecureEnterprise),
+		render.ProjectCalicoAPIServerTLSSecretName(operatorv1.Calico),
+	} {
+		if err = utils.AddSecretsWatch(c, secretName, common.OperatorNamespace()); err != nil {
+			return fmt.Errorf("failed to add watch for secret %s/%s: %w", common.OperatorNamespace(), secretName, err)
+		}
+	}
+
+	err = c.WatchObject(&operatorv1.Whisker{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("%s failed to watch primary resource: %w", controllerName, err)
+	}
+
+	if err = utils.AddInstallationWatch(c); err != nil {
+		return fmt.Errorf("%s failed to watch Installation resource: %w", controllerName, err)
+	}
+
+	if err = imageset.AddImageSetWatch(c); err != nil {
+		return fmt.Errorf("%s failed to watch ImageSet: %w", controllerName, err)
+	}
+
+	if err := utils.AddDeploymentWatch(c, whisker.WhiskerDeploymentName, whisker.WhiskerNamespace); err != nil {
+		return fmt.Errorf("%s failed to watch Whisker deployment: %w", controllerName, err)
+	}
+
+	// Watch for changes to TigeraStatus.
+	if err = utils.AddTigeraStatusWatch(c, ResourceName); err != nil {
+		return fmt.Errorf("clusterconnection-controller failed to watch management-cluster-connection Tigerastatus: %w", err)
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(
+	cli client.Client,
+	schema *runtime.Scheme,
+	statusMgr status.StatusManager,
+	p operatorv1.Provider,
+	tierWatchReady *utils.ReadyFlag,
+	opts options.AddOptions,
+) *Reconciler {
+	c := &Reconciler{
+		cli:            cli,
+		scheme:         schema,
+		provider:       p,
+		status:         statusMgr,
+		clusterDomain:  opts.ClusterDomain,
+		tierWatchReady: tierWatchReady,
+	}
+	c.status.Run(opts.ShutdownContext)
+	return c
+}
+
+// blank assignment to verify that ReconcileConnection implements reconcile.Reconciler
+var _ reconcile.Reconciler = &Reconciler{}
+
+// Reconciler reconciles a ManagementClusterConnection object
+type Reconciler struct {
+	cli            client.Client
+	scheme         *runtime.Scheme
+	provider       operatorv1.Provider
+	status         status.StatusManager
+	clusterDomain  string
+	tierWatchReady *utils.ReadyFlag
+}
+
+// Reconcile reads that state of the cluster for a Whisker object and makes changes based on the
+// state read and what is in the Whisker.Spec. The Controller will requeue the Request to be
+// processed again if the returned error is non-nil or Result.Requeue is true, otherwise upon completion it will
+// remove the work from the queue.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Whisker")
+	result := reconcile.Result{}
+
+	variant, installation, err := utils.GetInstallation(ctx, r.cli)
```

Comment:
I think we said we want to uninstall all of this if the Variant is enterprise since it's not supported there yet, right?